text/trace/0235-sampling-threshold-in-trace-state.md

```diff
@@ -0,0 +1,99 @@
+This allows for the continued expression of randomness using `r-value` as specified there using the key `r`.
+To distinguish the cases, this proposal uses the key `rv`.
+
+In order to make consistent sampling decisions across the entire path of the trace, two values SHOULD be propagated with the trace:
+
+1. A _random_ (or pseudo-random) 56-bit value, called `R` below.
+2. A 56-bit trace _threshold_ as expressed in the TraceState, called `T` below.
+
+The sampling decision is propagated with the following algorithm:
+
+* If the `th` key is not specified, Always Sample.
+* Else derive `T` by parsing the `th` key as a hex value as described below.
+* If `T` is 0 and the _sampled_ flag is set, Always Sample. This implies that non-probabilistic sampling is taking place.
+* Compare the 56 bits of `T` with the 56 bits of `R`. If `T <= R`, then do not sample.
+
+The `R` value MUST be derived as follows:
+
+* If the key `rv` is present in the Tracestate header, then `R = rv`.
+* Else if the Random Trace ID Flag is `true` in the traceparent header, then `R` is the lowest-order 56 bits of the trace-id.
+* Else `R` MUST be generated as a random value in the range `(0, (2**56)-1)` and added to the Tracestate header with key `rv`.
+
+The preferred way to propagate the `R` value is as the lowest 56 bits of the trace-id.
+If these bits are in fact random, the `random` trace-flag SHOULD be set as specified in [the W3C trace context specification](https://w3c.github.io/trace-context/#trace-id).
+There are circumstances where trace-id randomness is inadequate (for example, sampling a group of traces together); in these cases, an `rv` value is required.
+
+The value of the `rv` and `th` keys MUST be expressed as up to 14 hexadecimal characters from the set `[0-9a-f]`. Trailing zeros (but not leading zeros) may be omitted.
+
+Examples:
+`th` value is missing: Always Sample (probability = 100%). The AlwaysSample sampler in the OTel SDK should do this.
+`th=8` -- equivalent to `th=80000000000000`, which is 50% probability.
+`th=08` -- equivalent to `th=08000000000000`, which is 3.125% probability.
+`th=0` -- equivalent to `th=00000000000000`, which means Always Sample; this is outside of probabalistic sampling.
+
+The `T` value MUST be derived as follows:
+
+* If the `th` key is not present in the Tracestate header, then `T` is effectively 2^56 (which doesn't fit in 56 bits).
+* Else the value corresponding to the `th` key should be interpreted as above.
+
+Sampling Decisions SHOULD be propagated by setting the value of the `th` key in the Tracestate header according to the above.
+
+## Changing T and R values
+
+The T value MAY be modified.
+
+In the case of a downstream sampler -- a tail sampler on the collection path that is attempting to reduce the volume of traffic -- the sampler MAY modify the `th` header by reducing its value.
+It MAY NOT increase it, as it is not possible to retroactively adjust the sampling probability upward.
```

Comment:
why does anything need to be adjusted upward?

Reply:
Answer is, I think: 
It is illogical to adjust probability upward. It is never intentional, but a series of samplers can be misconfigured in a way that if this property were not checked, would allow misstating of sampling probability.

Under development in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24811, I have two sampler modes implemented as simple tail samplers. They are "equalizing" and "proportional". An equalizing sampler could be configured w/ an absolute threshold (e.g., 50% of all spans) while individual SDKs could be configured w/ 10% head sampling. When the 50%-equalizing tail sampler sees a 10%-head-sampled span, it will always pass the span through. It **must not** adjust the probability to 50% for the 10% head-sampled span, that's what this line is supposed to say.

With a proportional sampler, the resulting probabilities will be 50% and 5%.